File: Word-Formed/src/com/google/corrigan/owen/wordformed/HiScore.java
package com.google.corrigan.owen.wordformed;

import android.util.Log;

public class HiScore implements Comparable<HiScore>
	{
		private long id;
		private String name;
		private int score;

		public long getId() {
			return id;
		}

		public void setId(long id) {
			this.id = id;
		}

		public String getName() {
			return name;
		}

		public void setName(String name) {
			this.name = name;
		}

		// Will be used by the ArrayAdapter in the ListView
		@Override
		public String toString() {
			return name;
		}
		
		public void setScore(int score)
		{
			this.score = score;
		}
		
		public int getScore()
		{
			return score;
		}

		public int compareTo(HiScore other)
		{
			return other.score - score;
		}
	}

File: app/src/main/java/com/paddez/wordformed/HiScore.kt
package com.paddez.wordformed

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "hiscores")
data class HiScore(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val score: Int
) : Comparable<HiScore> {
    override fun compareTo(other: HiScore): Int {
        return other.score - score
    }

    override fun toString(): String {
        return name
    }
}
